pkg/neonvm/controllers/vm_controller.go

```diff
@@ -932,12 +933,26 @@ func runnerStatus(pod *corev1.Pod) runnerStatusKind {
 	case corev1.PodFailed:
 		return runnerFailed
 	case corev1.PodRunning:
-		return runnerRunning
+		return isRunnerPodReady(pod)
 	default:
 		panic(fmt.Errorf("unknown pod phase: %q", pod.Status.Phase))
 	}
 }
 
+// isRunnerPodReady returns whether the runner pod is ready respecting the readiness probe of its containers.
+func isRunnerPodReady(pod *corev1.Pod) runnerStatusKind {
+	if pod.Status.ContainerStatuses == nil {
```

Comment:
question: Is there a difference between `== nil` vs `len(...) == 0` ? I was a little surprised to see this check on a slice

Reply:
You right, I don't think there is any difference for a slice if we use `== nil` or `len(...) == 0`

Reply:
nil slices and non-nil empty slices are two different things: https://go.dev/play/p/UtIWK670wxW

I always use `len(...) == 0` to cover both cases.

Reply:
I usually tend use  != nil or len(slice) depending on the semantics of a particular code (like if I am interested in length or emptiness for sure) but I am ok with using any option here.